ports/zlib/vcpkg-cmake-wrapper.cmake

```
@@ -1,9 +1,8 @@
-set(ZLIB_ROOT "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}")
-find_path(ZLIB_INCLUDE_DIR NAMES zlib.h PATHS "${ZLIB_ROOT}/include" NO_DEFAULT_PATH)
-find_library(ZLIB_LIBRARY_RELEASE NAMES zlib  z PATHS "${ZLIB_ROOT}/lib" NO_DEFAULT_PATH)
-find_library(ZLIB_LIBRARY_DEBUG   NAMES zlibd z PATHS "${ZLIB_ROOT}/debug/lib" NO_DEFAULT_PATH)
-if(NOT ZLIB_INCLUDE_DIR OR NOT ZLIB_LIBRARY_RELEASE OR (NOT ZLIB_LIBRARY_DEBUG AND EXISTS "${ZLIB_ROOT}/debug/lib"))
-    message("Broken installation of vcpkg port zlib")
+find_path(ZLIB_INCLUDE_DIR NAMES zlib.h PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include" NO_DEFAULT_PATH)
+find_library(ZLIB_LIBRARY_RELEASE NAMES zlib  z PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib" NO_DEFAULT_PATH)
+find_library(ZLIB_LIBRARY_DEBUG   NAMES zlibd z PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib" NO_DEFAULT_PATH)
+if(NOT ZLIB_INCLUDE_DIR OR (NOT ZLIB_LIBRARY_RELEASE AND NOT ZLIB_LIBRARY_DEBUG))
+    message(FATAL_ERROR "Broken installation of vcpkg port zlib")
 endif()

```

```suggestion
find_library(ZLIB_LIBRARY_DEBUG   NAMES zlibd z PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib" NO_DEFAULT_PATH)
if(NOT ZLIB_INCLUDE_DIR OR NOT (ZLIB_LIBRARY_RELEASE OR ZLIB_LIBRARY_DEBUG))
    message(FATAL_ERROR "Broken installation of vcpkg port zlib")
endif()
```

Sorry for the longer suggestion block than necessary; GitHub wouldn't let me create it on the line. Can we demorgan out the inner condition to make it clearer that either missing the include directory or missing both release and debug is the erroneous condition.

---

i'd suggest something even easier. Since it's impossible to skip release and have debug-only, why not check for include & release?

---

Okay.
Now again Q2:
> Q2. If I have to change the PR again, shall I modernize portfile (quotes, deprecated functions), to save another rebuild for vcpkg users?

---

> i'd suggest something even easier. Since it's impossible to skip release and have debug-only, why not check for include & release?

I considered this option before but
- I was too frequently reminded of single config builds, even as a per-port option.
- I do not want to add more barriers for debug-only builds.
- zlib should be able to serve as a good example.

The only serious alternative is eventually offering function for frequently used wrapper patterns (select configurations, validate, add dependencies).